File: business-service/src/main/java/com/dayagent/entity/Parcel.java
package com.dayagent.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("parcel")
public class Parcel {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private String trackingNo;
    private String carrier;       // 快递公司，如"顺丰"、"京东"
    private String remark;        // 备注，如"耳机"
    private String status;        // 最新物流状态文本
    private String trackDetails;  // 完整物流轨迹 JSON
    private LocalDateTime lastChecked;
    private Boolean isDelivered;  // 是否已签收
    private LocalDateTime createdAt;
}

File: frontend/src/api/parcel.js
import api from './index.js'

export function addParcel(trackingNo, carrier, remark) {
  return api.post('/api/parcel', { trackingNo, carrier, remark }).then((r) => r.data)
}

export function getParcels() {
  return api.get('/api/parcel').then((r) => r.data)
}

export function deleteParcel(id) {
  return api.delete(`/api/parcel/${id}`).then((r) => r.data)
}
